### PR TITLE
54602/fix/contact form redux bug fixing

### PR DESCRIPTION
**Pull request by ornahasson**

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ import InteractedContactAccordion from './InteractedContactAccordion';
 import ContactQuestioningSchema from './ContactSection/Schemas/ContactQuestioningSchema';
 import { setIsLoading } from 'redux/IsLoading/isLoadingActionCreators';
 
-const SIZE_OF_CONTACTS = 4;//10;
```
Comment (chenFriedman) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx:22]: please remove coment

**Review by barnahum8 — APPROVED** [review]

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx [hunk 2]
```diff
@@ -65,28 +63,31 @@ const ContactQuestioning: React.FC<Props> = ({ id, isViewMode }: Props): JSX.Ele
     const listenScrollEvent = (event: React.UIEvent<HTMLDivElement>): void => {
         const element = event.target as HTMLElement;
         if (element.scrollHeight - element.scrollTop >= element.clientHeight && element.scrollHeight - element.scrollTop < element.clientHeight + 50) {
-            handleShowMoreContacts();
-        }
+                handleShowMoreContacts();
```
Comment (chenFriedman) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx:66]: please fix indent

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx [hunk 3]
```diff
@@ -65,28 +63,31 @@ const ContactQuestioning: React.FC<Props> = ({ id, isViewMode }: Props): JSX.Ele
...
-            loaded = SIZE_OF_CONTACTS;
-            loopWithSlice(0, SIZE_OF_CONTACTS);
-        }
-
+            if (interactedContacts.length>SIZE_OF_CONTACTS) {
+                loaded = SIZE_OF_CONTACTS;
+                setContactsToShow(interactedContacts.slice(0, SIZE_OF_CONTACTS));
+            }
+            else  {
+
```
Comment (barnahum8) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx:86]: Remove empty line

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx [hunk 4]
```diff
@@ -65,28 +63,31 @@ const ContactQuestioning: React.FC<Props> = ({ id, isViewMode }: Props): JSX.Ele
...
     useEffect(() => {
         loadInteractedContacts();
         loadFamilyRelationships();
         loadContactStatuses();
     }, []);
 
     useEffect(() => {
-
-        if (interactedContacts && interactedContacts.length > 0) {
+        if (interactedContacts && interactedContacts.length > 0 ){
```
Comment (chenFriedman) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx:79]: please fix alignment

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx (hunk 3 above)
Comment (chenFriedman) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioning.tsx:86]: please remove empty lines

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/ContactDetails.tsx [hunk 5]
```diff
@@ -47,7 +47,7 @@ const ContactDetails = (props: Props) => {
 
     const formStates = useSelector<StoreStateType, Map<number, FormStateObject>>(state => state.interactedContacts.formState);
 
-    const isFormInvalid = formStates?.get ? !formStates.get(interactedContact.id)?.isValid : false;
+    const isFormInvalid = JSON.stringify(errors) !== '{}' ? true : (formStates?.get ? !formStates.get(interactedContact.id)?.isValid : false);
```
Comment (chenFriedman) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/ContactDetails.tsx:50]: maybe you can change the condition to is empty, boolean or something nicer...

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/useReachContact.ts [hunk 6]
```diff
@@ -56,7 +58,7 @@ const useReachContact = (props: Props) => {
                         }
                     });
                 }
-                if (formHaveMissingFields && missingFieldsText !== '') {
+                if ((formHaveMissingFields && missingFieldsText !== '') ) {
```
Comment (chenFriedman) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/useReachContact.ts:61]: please remove extra ()

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/useReachContact.ts [hunk 7]
```diff
@@ -10,6 +10,8 @@ import ContactStatusCodes from 'models/enums/ContactStatusCodes';
 import GroupedInteractedContact from 'models/ContactQuestioning/GroupedInteractedContact';
 import { setInteractedContact } from 'redux/InteractedContacts/interactedContactsActionCreators';
 import interactedContactsReducer from 'redux/InteractedContacts/interactedContactsReducer';
+import InteractedContactFields from 'models/enums/InteractedContact';
```
Comment (chenFriedman) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo/useReachContact.ts:13]: I'm not sure why you need these 2 imports, I don't see any new use...

**Review by barnahum8 — DISMISSED** [review]

**Review by chenFriedman — CHANGES_REQUESTED** [review]

**Review by chenFriedman — APPROVED** [review]

**Pull request merged**